src/Components/LifecycleChart/LifecycleChart.tsx

```diff
@@ -343,6 +343,32 @@ const LifecycleChart: React.FC<LifecycleChartProps> = ({ lifecycleData }: Lifecy
     };
   }, [updatedLifecycleData.length]);
 
+  // Calculate padding based on the longest name
+  const calculateLeftPadding = () => {
+    if (updatedLifecycleData.length === 0) {
+      return 160; // Default padding if no data
+    }
+
+    // Get all names
+    const names = updatedLifecycleData.map((data) => data[0].x);
+
+    // Find the longest name
+    const longestName = names.reduce(
+      (longest, current) => (current.length > longest.length ? current : longest),
+      ''
+    );
+
+    // Calculate padding: base padding (60) + character count * character width factor
+    const charWidthFactor = 6;
+    const basePadding = 60;
+    const calculatedPadding = basePadding + longestName.length * charWidthFactor;
+
+    // Set a minimum and maximum boundary
+    return Math.max(160, Math.min(calculatedPadding, 250));
```

Comment:
```suggestion
    return Math.max(160, Math.min(calculatedPadding, 400));
```
I would increase a bit the minimum there, or maybe remove the `min` calculation completely and left just the `return Math.max(160, calculatedPadding);`.

I tried with `PostgreSQL Hello this is a really long name` and it was out of the borders with the `250` as the name was longer. Maybe it's a corner case, but I think this whole thing with long names a bit corner case :D

How I tested - changes in backend:
```
$ git diff
diff --git a/src/roadmap/v1/lifecycle/app_streams.py b/src/roadmap/v1/lifecycle/app_streams.py
index 03f9156..a816623 100644
--- a/src/roadmap/v1/lifecycle/app_streams.py
+++ b/src/roadmap/v1/lifecycle/app_streams.py
@@ -55,7 +55,7 @@ def get_display_name(app_stream: AppStreamEntity) -> str:
         "openjdk": "OpenJDK",
         "osinfo-db": "OSInfo DB",
         "php": "PHP",
-        "postgresql": "PostgreSQL",
+        "postgresql": "PostgreSQL Hello this is a really long name",
         "rhn-tools": "RHN Tools",
     }
     display_name = app_stream.name
```

Reply:
Sounds good!